app/components/layout/TabbedPage.js

```diff
@@ -116,7 +116,7 @@ function TabbedPage(props) {
       : "l2r";
     setDir(dir);
     setMatchedTab(matchedTab);
-  }, [ location ]);
+  }, [ location, previous, props ]);
```

Comment:
I don't think we need previous here, as if previous values are changing, it means that current values changed